backend/src/models/question.js

```diff
@@ -35,10 +33,21 @@ class Question {
      *                       contain the user data (probs eventually from an auth token)
      */
     postQuestion(courseID, { userID, title, body }) {
-        return db.insert('question', { courseID, userID, title, body })
+        return this.db
+            .insert('question', { courseID, userID, title, body })
     }
 }
 
-module.exports = (function(db) {
+let Singleton = null
+
+/**
+ * @param {object} db defaults to the db instance
+ */
+module.exports = function(db) {
+    if (!db) {
+        /* app environment, dev or prod */
+        return (Singleton = Singleton ? Singleton : new Question(require('./db'))) // eslint-disable-line
```

Comment:
How is singleton set in prod?

Reply:
Currently doesn't differentiate between prod/dev.